### PR TITLE
Fogl 634 argument parser, scheduler to start processes with address/port of core mgt, device service accepts cmd line args

### PR DESCRIPTION
**Pull request by ashwinscale**
Common argument parser Class foglamp/parser.py
Scheduler starts scheduled tasks with address, port, and name 
Device uses parser.py to parse command line arguments

Note that with this method, arguments can be indirectly via argparse without all arguments known beforehand.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,7 @@ docs/.cache/
 # Allure
 allure/
 allure-report/
+
+# storage specific
+build/
+services/storage/*
```
Comment (izoratti-dianomic) [inline .gitignore]: not sure about line 20, why should it be ignored?

File: src/python/foglamp/device/__main__.py [hunk 2]
```diff
@@ -16,25 +18,19 @@
...
-    plugin = 'CoAP'
+plugin = Parser.get('--name')
+core_mgt_port = Parser.get('--port')
+core_mgt_address = Parser.get('--address')
 
 if plugin is None:
-    # TODO: Output to STDERR
-    print("Required argument 'name' is missing")
-    # TODO: exit 1
+    print("Required argument '--name' is missing")
```
Comment (izoratti-dianomic) [inline src/python/foglamp/device/__main__.py]: maybe it does (so please ignore the msg), this should go to syslog, and so the other print(s)

File: src/python/foglamp/core/scheduler.py [hunk 3]
```diff
@@ -627,15 +632,24 @@ def max_running_tasks(self, value: int)->None:
         # the start of the awaited coroutine.
         args = self._process_scripts[schedule.process_name]
 
+        # add core management host and port to process script args
+        args_to_exec = args.copy()
+        args_to_exec.append("--port")
+        args_to_exec.append(str(self._core_management_port))
+        args_to_exec.append("--address")
+        args_to_exec.append("127.0.0.1")
```
Comment (izoratti-dianomic) [inline src/python/foglamp/core/scheduler.py]: @MarkRiddoch - just want to make sure this goes through socket file... does it?

File: src/python/foglamp/device/__main__.py [hunk 4]
```diff
@@ -16,25 +18,19 @@
...
-        plugin = None
-    else:
-        plugin = plugin
-except IndexError:
-    plugin = None
 
-if plugin is None:
-    # TODO: delete this
-    plugin = 'CoAP'
+plugin = Parser.get('--name')
```
Comment (MarkRiddoch) [inline src/python/foglamp/device/__main__.py]: The plugin is not passed in the name parameter the plugin comes form the configuration. The name is used to work out the name of the configuration section to look at https://docs.google.com/document/d/1c1dmMxH36IQNA_gFwxTPttyCeEStsrSxWYUqKeBbfNk/edit#heading=h.o2dnspmkfx4p

File: src/python/foglamp/core/scheduler.py (hunk 3 above)
Comment (MarkRiddoch) [reply src/python/foglamp/core/scheduler.py]: Yes, this uses the loopback interface and does not go down to the network hardware - I am guessing this is the question you are asking.

File: src/python/foglamp/parser.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python3
```
Comment (praveen-garg) [inline src/python/foglamp/parser.py]: remove this too

File: src/python/foglamp/parser.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+# FOGLAMP_END
+
+import argparse
+__author__ = "Ashwin Gopalakrishnan"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+
```
Comment (praveen-garg) [inline src/python/foglamp/parser.py]: add docstring and keep 2 lines only :)

File: src/python/foglamp/parser.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+class Parser(object):
+    @staticmethod
+    def get(key):
+      parser = argparse.ArgumentParser()
+      parser.add_argument(key)
+      parser.parse_known_args()
+      return list(vars(parser.parse_known_args()[0]).values())[0]
```
Comment (praveen-garg) [inline src/python/foglamp/parser.py]: it may have any known exception?

File: src/python/foglamp/parser.py [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+__version__ = "${VERSION}"
+
+
+
+class Parser(object):
+    @staticmethod
+    def get(key):
+      parser = argparse.ArgumentParser()
+      parser.add_argument(key)
+      parser.parse_known_args()
```
Comment (praveen-garg) [inline src/python/foglamp/parser.py]: is it being used as var in next line? returns a tuple or ?

File: .gitignore (hunk 1 above)
Comment (praveen-garg) [reply .gitignore]: I believe, we don't want to push static files to git (built via storage cmake/make)

File: src/python/foglamp/device/__main__.py [hunk 9]
```diff
@@ -16,25 +19,22 @@
...
-    print("Required argument 'name' is missing")
-    # TODO: exit 1
+    _logger.warning("Required argument '--name' is missing")
+elif core_mgt_port is None:
+    _logger.warning("Required argument '--port' is missing")
+elif core_mgt_address is None:
+    _logger.warning("Required argument '--address' is missing")
 else:
-    Server.start(plugin)
+    Server.start(plugin, core_mgt_port, core_mgt_address)
```
Comment (praveen-garg) [inline src/python/foglamp/device/__main__.py]: will be fixed in next PR; order of args

**Pull request merged**